gis4wrf/plugin/ui/widget_geo.py

```diff
@@ -21,41 +21,48 @@ def __init__(self, iface) -> None:
         self.options = get_options()
         self.msg_bar = MessageBar(iface)
 
-        vbox = QVBoxLayout()
-        self.setLayout(vbox)
-        
-        self.list_widget = QListWidget()
+        self.tree_widget = QTreeWidget ()
         self.populate_tree()
-        vbox.addWidget(self.list_widget)
 
         self.select_mandatory_hires_button = QPushButton('Select Mandatory Fields in Highest Resolution')
         self.select_mandatory_hires_button.clicked.connect(self.on_select_mandatory_hires_button_clicked)
-        vbox.addWidget(self.select_mandatory_hires_button)
 
         self.select_mandatory_lores_button = QPushButton('Select Mandatory Fields in Lowest Resolution')
         self.select_mandatory_lores_button.clicked.connect(self.on_select_mandatory_lores_button_clicked)
-        vbox.addWidget(self.select_mandatory_lores_button)
 
         self.download_button = QPushButton('Download Selected Datasets')
         self.download_button.clicked.connect(self.on_download_button_clicked)
-        vbox.addWidget(self.download_button)
 
         self.progress_bar = QProgressBar()
         self.progress_bar.setRange(0, 0)
         self.progress_bar.setTextVisible(False)
         self.progress_bar.hide()
+        
+        vbox = QVBoxLayout()
+        vbox.addWidget(self.tree_widget)
+        vbox.addWidget(self.select_mandatory_hires_button)
+        vbox.addWidget(self.select_mandatory_lores_button)
+        vbox.addWidget(self.download_button)
         vbox.addWidget(self.progress_bar)
+        self.setLayout(vbox)
 
     def populate_tree(self) -> None:
-        for name, label in geo_datasets.items():
-            item = QListWidgetItem('{}: {}'.format(label, name))
-            item.setData(Qt.UserRole, name)
-            if is_geo_dataset_downloaded(name, self.options.geog_dir):
+        self.tree_widget.setHeaderItem(QTreeWidgetItem([
+            'ID', 'Description', 'Resolution in °' ]))
+        self.tree_widget.setRootIsDecorated(False)
+        self.tree_widget.setSortingEnabled(True)
+        for id, (description, resolution) in  geo_datasets.items():
+            item = QTreeWidgetItem(self.tree_widget)
+            item.setText(0, id)
+            item.setData(0, Qt.UserRole, id)
+            item.setCheckState(0, Qt.Unchecked)
+            item.setText(1, description)
+            item.setText(2, str(resolution))
```

Comment:
This should be converted to what we have in the Data tab. Have a look at geogrid_tbl.py, the code within `if meta.proj_id == 'regular_ll':` and the `dd_to_dms` function should be pulled out into the util module so that it can be reused for this and your use case.

Reply:
I am not convinced we should change this -- by keeping this in decimal format it should make it easier for the user to go from domain resolution (decimal for lon/lat) to dataset resolution. Perhaps we should change the units in the geogrid_tbl.py actually. Alternatively, we may want allow the user to select between dd and dms but that is really a new feature/enhancement.

Reply:
Hm, I see your point. How about putting the dms format as tooltip only?

Reply:
Sounds good